tests/quantization/finegrained_fp8/test_fp8.py

```
@@ -197,6 +219,16 @@ def test_save_pretrained_multi_gpu(self):
             output = model.generate(**input_ids, max_new_tokens=self.max_new_tokens, do_sample=False)
             self.assertEqual(self.tokenizer.decode(output[0], skip_special_tokens=True), self.EXPECTED_OUTPUT)
 
+    def test_quantized_model_offload(self):
+        """
+        Simple test that checks if the quantized model returns an error when loading with cpu/disk offloaded
+        """
+        with self.assertRaisesRegex(
+            ValueError, "You are attempting to load an FP8 model with a device_map that contains a cpu/disk device."
+        ):
+            AutoModelForCausalLM.from_pretrained(
+                self.model_name, device_map=self.offload_device_map, quantization_config=self.quantization_config
+            )
```

this should raise an error with unquantized model but should actually work for quantized model. 

Also I think that we can actually make it work for unquantized model in the case we only have cpu offload but we can leave that for later.

---

Yes we could try that